server/oslc-ui-model/src/main/java/org/eclipse/lyo/server/ui/model/PreviewFactory.java

```diff
@@ -68,12 +68,25 @@ public static Preview getPreview(final AbstractResource aResource, List<String>
                     value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple, l);
                 } else {
                     Link link = (Link) getterMethod.invoke(aResource);
-                    value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple,
-                        constructLink(link));
+                    value = constructPropertyValue(PropertyDefintion.RepresentationType.LINK, multiple, constructLink(link));
                 }
             } else {
-                value = constructPropertyValue(PropertyDefintion.RepresentationType.TEXT, multiple,
-                    getterMethod.invoke(aResource));
+            	if (!isNotInlinedRepresentation) {
+                    if (multiple) {
+                        Collection<AbstractResource> rs = (Collection<AbstractResource>) getterMethod.invoke(aResource);
+                        List<org.eclipse.lyo.server.ui.model.Link> l = new ArrayList<>();
+                        for (AbstractResource r: rs) {
+                            l.add(constructLink(r.getAbout().toString(), r.toString()));
+                        }
```

Comment:
this can be rewritten with Java 8 stream in 1 line, eg https://mkyong.com/java8/java-8-streams-map-examples/

Reply:
of course.